interfaces/api/reloc/IKeyframeRetriever.h

```diff
@@ -53,26 +54,33 @@ class XPCF_IGNORE IKeyframeRetriever :
     virtual FrameworkReturnCode addKeyframe(const SRef<SolAR::datastructure::Keyframe> keyframe, bool useMatchedDescriptor=false) = 0;
 
 	/// @brief Suppress a keyframe from the retrieval model
-	/// @param[in] keyframe_id: the keyframe to supress from the retrieval model
+	/// @param[in] keyframeId: the keyframe to supress from the retrieval model
 	/// @return FrameworkReturnCode::_SUCCESS if the keyfram adding succeed, else FrameworkReturnCode::_ERROR_
-    virtual FrameworkReturnCode suppressKeyframe(uint32_t keyframe_id) = 0;
+    virtual FrameworkReturnCode suppressKeyframe(uint32_t keyframeId) = 0;
 
 
     /// @brief Retrieve a set of keyframes close to the frame pass in input.
     /// @param[in] frame: the frame for which we want to retrieve close keyframes.
-    /// @param[out] retKeyframes_id: a set of keyframe ids which are close to the frame pass in input
+    /// @param[out] retKeyframeId: a set of keyframe ids which are close to the frame pass in input
     /// @return FrameworkReturnCode::_SUCCESS if the retrieve succeed, else FrameworkReturnCode::_ERROR_
     virtual FrameworkReturnCode retrieve(const SRef<SolAR::datastructure::Frame> frame,
-                                         std::vector<uint32_t> & retKeyframes_id) = 0;
+                                         std::vector<uint32_t> & retKeyframeId) = 0;
+
+    /// @brief Retrieve a set of keyframes close to the frame passed in input.
+    /// @param[in] frame: the frame for which we want to retrieve close keyframes.
+    /// @param[out] retKeyframeIdScore: a set of pairs of keyframe id and the corresponding retrieval score (sorted by score in decreasing order)
+    /// @return FrameworkReturnCode::_SUCCESS if the retrieve succeed, else FrameworkReturnCode::_ERROR_
+    virtual FrameworkReturnCode retrieve(const SRef<SolAR::datastructure::Frame> frame,
+                                         std::vector<std::pair<uint32_t, double>> & retKeyframeIdScore) = 0;
 
 	/// @brief Retrieve a set of keyframes close to the frame pass in input.
 	/// @param[in] frame: the frame for which we want to retrieve close keyframes.
-    /// @param[in] canKeyframes_id: a set including id of keyframe candidates
-	/// @param[out] retKeyframes_id: a set of keyframe ids which are close to the frame pass in input
+    /// @param[in] canKeyframeId: a set including id of keyframe candidates
+	/// @param[out] retKeyframeId: a set of keyframe ids which are close to the frame pass in input
 	/// @return FrameworkReturnCode::_SUCCESS if the retrieve succeed, else FrameworkReturnCode::_ERROR_
     virtual FrameworkReturnCode retrieve(const SRef<SolAR::datastructure::Frame> frame,
-                                         const std::set<unsigned int> & canKeyframes_id,
-                                         std::vector<uint32_t> & retKeyframes_id) = 0;
+                                         const std::set<unsigned int> & canKeyframeId,
```

Comment:
Also, I don't know for the other reviewers, but I was a bit confused by the variable name, I wondered what the `can`  prefix stood for ("can" as in "they can be valid ids" ?), but I'm guessing it's for `candidates` ?
If you think `candidateKeyframeIds` is too long, what about removing `keyframe` from the names, as it may be inferred from context (we're in a KeyframeRetriever, it can be described in details in the doxygen, the method signature `retrieve(frame, retrievedIds)`....)
So it would become:
- `canKeyframeId` -> `candidateIds`
- `retKeyframeId` -> `retrievedIds`

(plural since it's a collection)

Reply:
Thanks JM for the suggestion ! yes I agree with you that it would be better to use uint32_t and rename some arguments here. modifs will be made to both solarframework and solarmodulefbow